[DependencyInjection] Add docs for default priority method for tagged services (#12697)
symfony/symfony-docs: service_container/tags.rst

        }
    }

> Prioritizing Tagged Services

Reviewer: Maybe „Tagged Services with Priority“ sounds more natural?

cc @javiereguiluz

Author: For me the current one sounds a bit clearer, but if others think that "Tagged Services with Priority" is better I will just update it.

Author: Changed it to how you suggested

~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> 

Reviewer 2: We need a `.. versionadded:: 4.4` here, see https://symfony.com/doc/current/contributing/documentation/format.html#new-features-behavior-changes-or-deprecations.

Author: Done

Author: Added it. Thanks :)

Reviewer 2: Missing a dot :)

Author: Added

> The tagged services can be prioritized using the ``priority`` attribute:

Reviewer: To me a sentence is missing what I can achieve by using priority. I mean we are talking a lot about the how but not the why. A real world use case would be very helpful to get context. Thanks

Author: From my side it looks pretty obvious, but maybe I am not the best person to ask since I did not need this until now. I started looking into this on Symfony Hack Day. If you can share some use case, we can surely added it.

Reviewer 2: What about expanding that sentence a little? Something like:
```rst
The tagged services can be prioritized using the ``priority`` attribute,
so a consumer can be injected a sorted collection:
```

Author: Updated. I tried to avoid using the word `consumer` since that can have different meanings and might confuse people. Thanks @HeahDude

                arguments:
                    - !tagged_iterator { tag: app.handler, default_priority_method: getPriority }

    .. code-block:: xml

        <!-- config/services.xml -->
        <?xml version="1.0" encoding="UTF-8" ?>
        <container xmlns="http://symfony.com/schema/dic/services"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://symfony.com/schema/dic/services
                https://symfony.com/schema/dic/services/services-1.0.xsd">
            <services>
                <service id="App\HandlerCollection">
                    <argument type="tagged" tag="app.handler" default-priority-method="getPriority"/>
                </service>
            </services>
        </container>

    .. code-block:: php

        // config/services.php
        namespace Symfony\Component\DependencyInjection\Loader\Configurator;

        use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;

        return function (ContainerConfigurator $configurator) {
            $services = $configurator->services();

            // ...

            $services->set(App\HandlerCollection::class)

>                 ->args(

Reviewer 2: ```suggestion
                ->args([
                    new TaggedIteratorArgument(...),
                ])
```

        }
    }

> Prioritizing tagged services

Reviewer: ```suggestion
Prioritizing Tagged Services
```

> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

The tagged services can be prioritized using the ``priority`` attribute:

.. configuration-block::

    .. code-block:: yaml

        # config/services.yaml
        services:
            App\Handler\One:
                tags:
                    - { name: 'app.handler', priority: 20 }

    .. code-block:: xml

        <!-- config/services.xml -->
        <?xml version="1.0" encoding="UTF-8" ?>
        <container xmlns="http://symfony.com/schema/dic/services"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://symfony.com/schema/dic/services
                https://symfony.com/schema/dic/services/services-1.0.xsd">

            <services>
                <service id="App\Handler\One">
                    <tag name="app.handler" priority="20"/>
                </service>
            </services>
        </container>

    .. code-block:: php

        // config/services.php
        $container->register(App\Handler\One::class)
            ->addTag('app.handler', ['priority' => 20]);

> Note that any other custom attributes will be ignored by this feature.

Reviewer: ```suggestion
.. note::

    Note that any other custom attribute will be ignored by this feature.
```

> Another option, which is particularly useful when using autoconfiguring tags, is to implement the ``getDefaultPriority`` static method on a collected class::

Reviewer: ```suggestion
Another option, which is particularly useful when using autoconfiguring tags, is to implement the
static ``getDefaultPriority`` method on the service itself::
```

    // src/App/Handler/One.php
    namespace App/Handler;

    class One
    {
        public static function getDefaultPriority(): int
        {
            return 3;
        }
    }

> If you want to have another method defining the priority, can define it in the configuration of the collecting service:

Reviewer: ```suggestion
If you want to have another method defining the priority, you can define it in the configuration of the collecting service:
```

.. configuration-block::

    .. code-block:: yaml

        # config/services.yaml
        services:
            App\HandlerCollection:
                # inject all services tagged with app.handler as first argument
                arguments:
                    - !tagged_iterator { tag: app.handler, default_priority_method: getPriority }

    .. code-block:: xml

        <!-- config/services.xml -->
        <?xml version="1.0" encoding="UTF-8" ?>
        <container xmlns="http://symfony.com/schema/dic/services"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://symfony.com/schema/dic/services
                https://symfony.com/schema/dic/services/services-1.0.xsd">

> 

Reviewer: ```suggestion
```

                    - !tagged_iterator { tag: app.handler, default_priority_method: getPriority }

    .. code-block:: xml

        <!-- config/services.xml -->
        <?xml version="1.0" encoding="UTF-8" ?>
        <container xmlns="http://symfony.com/schema/dic/services"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://symfony.com/schema/dic/services
                https://symfony.com/schema/dic/services/services-1.0.xsd">
            <services>
                <service id="App\HandlerCollection">
                    <argument type="tagged" tag="app.handler" default-priority-method="getPriority"/>
                </service>
            </services>
        </container>

    .. code-block:: php

        // config/services.php
        namespace Symfony\Component\DependencyInjection\Loader\Configurator;

        use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;

        return function (ContainerConfigurator $configurator) {
            $services = $configurator->services();

            // ...

            $services->set(App\HandlerCollection::class)
                ->args([

>                         new TaggedIteratorArgument('app.handler', null, null, false, 'getPriority'),

Reviewer 2: ```suggestion
                        tagged_iterator('app.handler', null, null, 'getPriority'),
```

                    ]

>                 );

Reviewer 2: Semi colon on next line

Author: Fixed
